### ISSUE TITLE
Pointer events emit as touch events in 4.5.2

### ISSUE THREAD

**Issue by stephenweaver** [pixijs/pixi.js#4040]
I'll make a jsfiddle later, but wanted to go ahead and report it.

EDITED:
4.5.1: http://requirebin.com/?gist=171f18b9ef624711a287db036a755d49
4.5.2: http://requirebin.com/?gist=79b98d57d6a51b11f9d076affc0efef4

Notice that in 4.5.2, there is no pointer id which makes it very hard to track the points if there's more than one...doesn't look like it's possible unless i'm missing something

So I guess it would be more accurate to say that multi-touch is broken. But my site code was counting always having a pointerId. It would be nice if originalEvent was actually the original event.

**Comment by bigtimebuddy** [comment]
Need more details. Cannot reproduce on iOS (http://pixijs.github.io/examples/index.html#/demos/interactivity.js)

**Comment by stephenweaver** [comment]
That demo is running 4.5.1. But yeah, I'll put a jsfiddle together tomorrow.

**Comment by stephenweaver** [comment]
Updated the description to include some demos

**Comment by andrewstart** [comment]
In 4.5.2 we made a change to always use TouchEvents instead of PointerEvents, because of an expectation that `evt.data.originalEvent` would be a TouchEvent. Unfortunately, using both PointerEvents and TouchEvents would make it very complicated to track things based on pointer id, because that would likely be shared between the events.

However, we also expose most/all relevant event data for each event/touch now - if you want to work with the PointerEvent API to get your data, then that should all exist on `evt.data`, including `pointerId`, regardless of if the original event is a TouchEvent, PointerEvent, or MouseEvent.

**Comment by stephenweaver** [comment]
I guess I'm just confused why if i have the option to do listen to pointer event, the pointer events would give me touch events. 

This is what we were doing. If the browser supports pointers, then use them. Otherwise use mouse and touch events. 

`  if (window.PointerEvent) {
    this.on(Events.pointerdown, this.onPointerDown);
    this.on(Events.pointerup, this.onPointerUp);
    this.on(Events.pointerupoutside, this.onPointerUp);
    this.on(Events.pointerover, this.onMouseOver);
    this.on(Events.pointerout, this.onMouseOut);
  } else {
    this.on(Events.mouseover, this.onMouseOver);
    this.on(Events.mouseout, this.onMouseOut);
    this.on(Events.mousedown, this.onMouseDown);
    this.on(Events.mouseup, this.onMouseUp);
    this.on(Events.mouseupoutside, this.onMouseUp);
    this.on(Events.touchstart, this.onTouchStart);
    this.on(Events.touchend, this.onTouchEnd);
    this.on(Events.touchendoutside, this.onTouchEnd);
  }`

Now that you tell me it's giving touch events when it should be giving pointer events, I now see the odd side effect, which is why I was so confused yesterday. If I log the event from say pointerdown and drill down into the object (event.data.originalEvent) in the console, i see that it looks like a PointerEvent; however, if I access or console.log(evt.data.originalEvent), it's a Touch event?! I find this pretty unpredictable, but does explain why i'm getting undefined when trying to grab the pointerId, even though I can see it when logging the event and drilling down into it.

So I guess I can make it the site work with this, i just don't like it. On a side not, to do this and only rev the hotfix version is pretty bold.

**Comment by andrewstart** [comment]
Ever since we first implemented PointerEvents, we also fired `pointerup` and friends on TouchEvents and MouseEvents so that devs using PIXI could switch over to just pointer events, instead of the big if/else, and we would shield them from variations in browser support.

**Comment by stephenweaver** [comment]
Don't get me wrong, I'm glad that pixi now supports pointer events. In 2015, when I started the app, I added pointer events by overriding some of the methods in the InteractionManager.js that looked something like this (https://github.com/pixijs/pixi.js/pull/2195 at the time of the pr, there were some issues). So I'm glad that I don't have to override anything anymore!

I'm not against normalizing events and adding new properties to InteractionData. All I'm asking is that the originalEvent not be modified, and this change happened between 4.5.1 and 4.5.2.

I hope I'm clearly conveying what I'm trying to say. I'm done going back and forth on though. If I'm the only one thinks that the originalEvent shouldn't be modified, then feel free to close my ticket.
